cylc/flow/xtrigger_mgr.py

```diff
@@ -435,18 +509,26 @@ def call_xtriggers_async(self, itask: 'TaskProxy'):
             itask: task proxy to check.
         """
         for label, sig, ctx, _ in self._get_xtrigs(itask, unsat_only=True):
-            # Special case: quick synchronous clock check:
-            if sig.startswith("wall_clock"):
+            if label in self.wall_clock_labels:
+                # Special case: quick synchronous clock check.
                 if sig in self.sat_xtrig:
                     # Already satisfied, just update the task
                     itask.state.xtriggers[label] = True
                 elif wall_clock(*ctx.func_args, **ctx.func_kwargs):
-                    # Newly satisfied
                     itask.state.xtriggers[label] = True
                     self.sat_xtrig[sig] = {}
                     self.data_store_mgr.delta_task_xtrigger(sig, True)
                     self.workflow_db_mgr.put_xtriggers({sig: {}})
                     LOG.info('xtrigger satisfied: %s = %s', label, sig)
+                    if (
+                        itask.is_xtrigger_sequential
+                        and all(
+                            itask.state.xtriggers[seq_label]
+                            for seq_label in itask.state.xtriggers
+                            if seq_label in self.sequential_xtrigger_labels
+                        )
+                    ):
+                        self.sequential_spawn_next.add(itask.identity)
```

Comment:
```diff
diff --git a/cylc/flow/xtrigger_mgr.py b/cylc/flow/xtrigger_mgr.py
index 90c32b8dc..861fc791e 100644
--- a/cylc/flow/xtrigger_mgr.py
+++ b/cylc/flow/xtrigger_mgr.py
@@ -521,12 +521,5 @@ class XtriggerManager:
                     self.workflow_db_mgr.put_xtriggers({sig: {}})
                     LOG.info('xtrigger satisfied: %s = %s', label, sig)
-                    if (
-                        itask.is_xtrigger_sequential
-                        and all(
-                            itask.state.xtriggers[seq_label]
-                            for seq_label in itask.state.xtriggers
-                            if seq_label in self.sequential_xtrigger_labels
-                        )
-                    ):
+                    if self.all_task_seq_xtriggers_satisfied(itask):
                         self.sequential_spawn_next.add(itask.identity)
                     self.do_housekeeping = True
@@ -548,12 +541,5 @@ class XtriggerManager:
                             xtrigger_env
                         )
-                    if (
-                        itask.is_xtrigger_sequential
-                        and all(
-                            itask.state.xtriggers[seq_label]
-                            for seq_label in itask.state.xtriggers
-                            if seq_label in self.sequential_xtrigger_labels
-                        )
-                    ):
+                    if self.all_task_seq_xtriggers_satisfied(itask):
                         self.sequential_spawn_next.add(itask.identity)
                 continue
@@ -613,2 +599,10 @@ class XtriggerManager:
             self.sat_xtrig[sig] = results
             self.do_housekeeping = True
+
+    def all_task_seq_xtriggers_satisfied(self, itask: 'TaskProxy') -> bool:
+        """Check if all sequential xtriggers are satisfied for a task."""
+        return itask.is_xtrigger_sequential and all(
+            itask.state.xtriggers[label]
+            for label in itask.state.xtriggers
+            if label in self.sequential_xtrigger_labels
+        )
```

Reply:
Yeah, thought about doing this.. But wasn't sure it mattered, but that method may potentially get called elsewhere too..
Done now.

Reply:
If it's used twice that's enough to make it its own method for me 👍  Also for readability